packages/block-editor/src/store/actions.js

```diff
@@ -211,15 +257,27 @@ export function stopMultiSelect() {
  *
  * @param {string} start First block of the multi selection.
  * @param {string} end   Last block of the multiselection.
- *
- * @return {Object} Action object.
  */
-export function multiSelect( start, end ) {
-	return {
+export function* multiSelect( start, end ) {
+	yield {
 		type: 'MULTI_SELECT',
 		start,
 		end,
 	};
+
+	const blockCount = yield controls.select(
+		'core/block-editor',
+		'getSelectedBlockCount'
+	);
+
+	speak(
```

Comment:
It isn't clear when we use controls in actions. Technically speaking `speak` is also a side effect, although it definitely isn't something that you need to wait for. I see that it's inlined the same way in other places so it might be may misunderstanding when to use controls.